src/nmea.c

```diff
@@ -104,10 +118,10 @@ void nmea_gpgga(const double pos_llh[3], const gps_time_t *gps_t, u8 n_used,
 
   double frac_s = fmod(gps_t->tow, 1.0);
 
-  s16 lat_deg = (s16)((180.0 / M_PI) * pos_llh[0]);
-  double lat_min = fabs(60 * ((180.0 / M_PI) * pos_llh[0] - lat_deg));
-  s16 lon_deg = (s16)((180.0 / M_PI) * pos_llh[1]);
-  double lon_min = fabs(60 * ((180.0 / M_PI) * pos_llh[1] - lon_deg));
+  s16 lat_deg = (s16)R2D * (pos_llh[0]);
```

Comment:
Did you test this again? I think the casting will cause problems here. I think it should be:
`s16 lat_deg = R2D * pos_llh[0];`